plugins/database/mysql/connection_producer.go

```diff
@@ -110,6 +125,34 @@ func (c *mySQLConnectionProducer) Init(ctx context.Context, conf map[string]inte
 		mysql.RegisterTLSConfig(c.tlsConfigName, tlsConfig)
 	}
 
+	if c.AuthType == authTypeGCPIAM {
+		c.cloudDriverName, err = uuid.GenerateUUID()
+		if err != nil {
+			return nil, fmt.Errorf("unable to generate UUID for IAM configuration: %w", err)
+		}
+
+		credentials := c.Credentials
+
+		// for _most_ sql databases, the driver itself contains no state. In the case of google's cloudsql drivers,
+		// however, the driver might store a credentials file, in which case the state stored by the driver is in
+		// fact critical to the proper function of the connection.
+		//
+		// This poses one big obvious problem - each configured cloud database might/will need its own driver registration,
+		// the name of which we have to track, and even worse in the MySQL case, that name is also the name of the
+		// dialer that is registered by MySQL in the dsn: protocol in the DSN acts double duty if a custom dialer
+		// is registered. This means that either we can't hide the name of the dialer from the user OR we have to rewrite
+		// the DSN after the user provides it. We already do this, KIND OF for the TLS config, but this modification
+		// is much more dramatic.
+		//
+		// Another potentially latent issue is that it is difficult to clear out the driver (or its underlying dialer)
+		// if the config changes or if the backend is deleted. This has the potential to become a bit of an issue if
+		// somehow millions upon millions of configs are written.
```

Comment:
What do we mean by "difficult to clear out the driver?" Are we not able to handle config updates or backend deletion?

Reply:
Drivers that are registered (via `sql.RegisterDriver`) can't really be unregistered so since we need a Driver/Dialer per IAM identity, I'm not sure a delete totally clears out that driver struct. We can `close()` the Dialer which should clean up a lot of things, but `sql.RegisterDriver` we can't completely undo.